src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -100,6 +115,108 @@ class StreamReader {
     [[nodiscard]] virtual auto decode_range(size_t begin_idx, size_t end_idx, bool use_filter) const
             -> DecodedResultsTsType = 0;
 
+    /**
+     * A templated function that implements `filter_log_events` for both
+     * `UnstructuredIrStreamReader` and `StructuredIrStreamReader`.
+     *
+     * @param log_level_filter
+     * @param log_events Derived class's log events.
+     * @param[out] filtered_log_event_map A reference to derived class's `FilteredLogEventsMap`
+     * that stores filtered result.
```

Comment:
```suggestion
     * @param log_events
     * @param[out] filtered_log_event_map Returns the filtered log events.
```

We use "Returns" for the parameter according to our [internal guidelines](https://www.notion.so/yscope/WIP-Coding-Guidelines-9a308b847a5343958ba3cb97a850be66?pvs=4#9c18b7d2cf2543dea6a3b56f05072287).